rust/scx_utils/src/topology.rs

```diff
@@ -460,13 +474,26 @@ fn create_insert_cpu(
     let base_freq = read_file_usize(&freq_path.join("base_frequency")).unwrap_or(max_freq);
     let trans_lat_ns = read_file_usize(&freq_path.join("cpuinfo_transition_latency")).unwrap_or(0);
 
-    let llc = node.llcs.entry(llc_id).or_insert(Arc::new(Llc {
-        id: llc_id,
+    if node.llc_kernel_ids.len() == 0 && new_node {
+        *llc_id += 1;
+    }
+
+    let inc_llc = if node.llc_kernel_ids.len() == 0 || node.llc_kernel_ids.contains(&llc_kernel_id)
+    {
+        false
+    } else {
+        true
+    };
+    node.llc_kernel_ids.insert(llc_kernel_id);
```

Comment:
Is this correct? Let's say there are four cpus 0 1 2 3. Kernel says 0 2 are on llc 0 and 1 3 or llc 4. We want to assign llc 0 to cpu 0 2 and llc 1 to 1 3, right? Wouldn't the above code do that?